pkg/deploy/util/util.go

```diff
@@ -416,3 +444,16 @@ func (s ByMostRecent) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s ByMostRecent) Less(i, j int) bool {
 	return !s[i].CreationTimestamp.Before(s[j].CreationTimestamp)
 }
+
+// ControllersByCreationTimestampAsc sorts a list of ReplicationControllers by creation timestamp oldest first, using their names as a tie breaker.
+type ControllersByCreationTimestampAsc []api.ReplicationController
```

Comment:
We shouldn't cleanup rcs with replicas and if the current deployment has 0
replicas and gets cleaned up (unlikely), it will be recreated on the spot
anyway. We assume that annotations work in all other cases, what's special
about not doing it here?

On Mon, Jul 11, 2016 at 6:01 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> In pkg/deploy/util/util.go
> <https://github.com/openshift/origin/pull/8691#discussion_r70286230>:
>
> > @@ -416,3 +444,16 @@ func (s ByMostRecent) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
> >  func (s ByMostRecent) Less(i, j int) bool {
> >  	return !s[i].CreationTimestamp.Before(s[j].CreationTimestamp)
> >  }
> > +
> > +// ControllersByCreationTimestampAsc sorts a list of ReplicationControllers by creation timestamp oldest first, using their names as a tie breaker.
> > +type ControllersByCreationTimestampAsc []api.ReplicationController
>
> Looking at the logic for DeploymentVersionFor, what guarantees can we
> make about the versions returned? When would an RC have a malformed version
> annotation? Can we take those risks when cleaning up? Are we at risk of
> "cleanung up" the current deployment if someone's broken that annotation on
> it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8691/files/9170bf7c4858c4843e68a3f980f568cb55511168#r70286230>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADuFf43ZwPZj50C-ZYditosdo4f5eTtuks5qUmi-gaJpZM4ITGf0>
> .
>